### ISSUE TITLE
Not reloading in Firefox in Responsive Design Mode

### ISSUE THREAD

**Issue by lorenzos** [twolfson/livereload-extensions#5]
Steps to reproduce:

1. Enable Live Reload normally with the toolbar icon
2. Edit something, Firefox reloads the page or style
3. Hit `CTRL+SHIFT+M` to get into Responsive Design Mode
4. Edit something that will cause "hard" reloading, the browser reloads page once
5. Edit again, the browser doesn't reload the page, nor styles anymore
6. Hit `CTRL+SHIFT+M` to exit from Responsive Design Mode
7. Edit again, it's not working anymore

**Comment by twolfson** [comment]
I've successfully reproduced this error but don't have time to dive into it fully until later this week/weekend. There seems to be some helpful messages in `about:debugging` which lead me to believe it might be a Firefox issue

Here's a set of normal errors when switching into Responsive Design Mode:

```
TypeError: zoom is undefined[Learn More]  viewZoomOverlay.js:47:5
TypeError: win.gBrowser is undefined[Learn More]  ZoomUI.jsm:69:1
win.gBrowser is undefined  ReaderParent.jsm:70
Type error for parameter details (Error processing tabId: Expected integer instead of {"tabId":17,"windowId":3}) for browserAction.setTitle. global.js:335
```

When we attempt to enable LiveReload within Responsive Design Mode, we get the following error (normally don't get it, have verified server is running):

```
Connecting to ws://127.0.0.1:35729/livereload...  global.js:213:5
Web socket connected, sending handshake.  global.js:226:9
Incoming message: {"serverName": "livereload-tornado", "command": "hello", "protocols": ["http://livereload.com/protocols/official-7"]}  global.js:247:9
Error: Could not establish connection. Receiving end does not exist.  undefined
```

When do enable our LiveReload extension before entering Responsive Design Mode and trigger a hard reload, we get a new error message:

```
Error: sender.tab is undefined  undefined
```

These 2 scenarios could be related or be different but in any case, I'm going to leave this issue as "help wanted" for now. I'll comment if/when I resume work on it

**Comment by twolfson** [comment]
Looking into this again now

**Comment by twolfson** [comment]
Alright, I've gotten to the bottom of it. Going to try to build a small reproducible case for the Mozilla team. Here's the summary:

- In Responsive Design Mode, both `browserAction.setTitle` and `browserAction.setIcon` error out saying they expect an integer instead of an object. This contradicts the documentation as well as behavior working fine in normal mode
- In Responsive Design Mode, `runtime.onMessage` doesn't pass optional `sender.tab` despite being in a tab. This contradicts behavior in normal mode
- In Responsive Design Mode, `tabs.sendMessage` doesn't identify tabs existence despite being the value set via `browserAction.onClicked`. This contradicts behavior in normal mode

**Comment by twolfson** [comment]
1 down: `runtime.onMessage` https://bugzilla.mozilla.org/show_bug.cgi?id=1425657

2 to go

**Comment by twolfson** [comment]
Oooh, it turns out the `browserAction` but was misidentified and is a bug in our code. We were expecting the wrong arguments for `tabs.onActivated`. I'll submit a fix for that shortly. Now the only one left is `tabs.sendMessage`

**Comment by twolfson** [comment]
Alright, filed `tabs.onActivated`: https://bugzilla.mozilla.org/show_bug.cgi?id=1425660

Going to remove "help wanted" and mark this issue as externally blocked

**Comment by twolfson** [comment]
I've cut a new release `20.2.0` with the `tabs.onActivated` fix 👍

**Comment by lorenzos** [comment]
Thanks for the updates!

**Comment by twolfson** [comment]
It looks like both of the issues were already resolved in Firefox Beta 58.0. You should be good to use Responsive Design Mode if you switch over to Firefox Beta, Firefox Developer Edition, or Firefox Nightly

I'm eagerly closing this issue but feel free to re-open it if Firefox Beta 58.0 doesn't work
